lib/getBinaryGltf.js

```diff
@@ -118,12 +118,15 @@ function getBinaryGltf(gltf, embed, embedImage) {
     // Remove extras objects before writing
     removePipelineExtras(gltf);
 
-    // Create padded binary scene string and calculate total length
+    // Create padded binary scene buffer and calculate total length
     var sceneString = JSON.stringify(gltf);
-    var sceneLength = Buffer.byteLength(sceneString);
-    sceneLength += 4 - (sceneLength % 4);
-    var padding = new Array(sceneLength + 1).join(' ');
-    sceneString = (sceneString + padding).substring(0, sceneLength);
+    var sceneBuffer = new Buffer(sceneString);
+    var sceneLength = sceneBuffer.length;
+    var paddingLength = 4 - (sceneLength % 4); // pad out to 4 bytes as per glb specification
+    sceneLength += paddingLength;
+    var paddingBuffer = new Buffer(paddingLength).fill(' ');
```

Comment:
Use `Buffer.alloc(paddingLength, ' ')` instead. We should avoid `new Buffer` throughout but that can be a sweeping change handled later.